nautobot/cloud/views.py

```diff
@@ -23,6 +33,58 @@ class CloudAccountUIViewSet(NautobotUIViewSet):
     form_class = CloudAccountForm
 
 
+class CloudNetworkUIViewSet(NautobotUIViewSet):
+    queryset = CloudNetwork.objects.all()
+    filterset_class = CloudNetworkFilterSet
+    filterset_form_class = CloudNetworkFilterForm
+    serializer_class = CloudNetworkSerializer
+    table_class = CloudNetworkTable
+    form_class = CloudNetworkForm
+    bulk_update_form_class = CloudNetworkBulkEditForm
+
+    def get_extra_context(self, request, instance=None):
+        context = super().get_extra_context(request, instance)
+        if self.action == "retrieve":
+            prefixes = instance.prefixes.restrict(request.user, "view")
+            prefix_count = prefixes.count()
+            prefix_table = PrefixTable(prefixes.select_related("namespace"))
+            children_table = CloudNetworkTable(instance.children.all())
+
+            circuits = (
+                Circuit.objects.restrict(request.user, "view")
+                .filter(
+                    Q(circuit_termination_a__cloud_network=instance.pk)
+                    | Q(circuit_termination_z__cloud_network=instance.pk)
+                )
+                .select_related("circuit_type", "tenant")
+                .prefetch_related("circuit_terminations__location")
```

Comment:
May not be needed as `BaseTable` now does automatic select_related/prefetch_related determination?